org.lflang/src/org/lflang/ASTUtils.java

```diff
@@ -432,15 +432,17 @@ private static Instantiation getDelayInstance(Reactor delayClass,
                     WidthTerm term = factory.createWidthTerm();
                     term.setPort(EcoreUtil.copy(port));
                     widthSpec.getTerms().add(term);
-                }   
+                }
             } else {
                 widthSpec.setOfVariableLength(true);
             }
             delayInstance.setWidthSpec(widthSpec);
         }
         Assignment assignment = factory.createAssignment();
         assignment.setLhs(delayClass.getParameters().get(0));
-        assignment.getRhs().add(delay);
+        Initializer init = factory.createInitializer();
+        init.getExprs().add(connection.getDelay());
```

Comment:
Maybe we should assert the following
```suggestion
        init.getExprs().add(Objects.requireNonNull(connection.getDelay(), "null delay"));
```